ui/app/controllers/vault/cluster/secrets/backends.js

```diff
@@ -28,28 +28,20 @@ export default class VaultClusterSecretsBackendController extends Controller {
     if (this.selectedEngineType) {
       // check first if the user has also filtered by name.
       if (this.selectedEngineName) {
-        return this.filterByName(sortedBackends);
+        return sortedBackends.filter((backend) => this.selectedEngineName === backend.get('id'));
```

Comment:
I don't think we need to use `.get` anymore, we could instead do: 
```suggestion
        return sortedBackends.filter((backend) => this.selectedEngineName === backend.id);
```
But if that doesn't work for some reason I'd like to understand why!